### PR TITLE
cmd/evm: calc base fee if parent data is present

### PR DESCRIPTION
**Pull request by lightclient**
Currently, in order to chain together sequential valid `t8n` transitions the caller must manually calculate the block base fee. This PR adds support for the necessary parent fee market data to calculate the base fee for the current transition. 

Concretely, `env` is extended to accept the following:

* `parentBaseFee`
* `parentGasUsed`
* `parentGasLimit`

Example usage can be found in `./cmd/evm/testdata/25`.

### CODE REVIEW

File: cmd/evm/internal/t8ntool/transition.go [hunk 1]
```diff
@@ -247,7 +248,15 @@ func Transition(ctx *cli.Context) error {
 	}
 	// Sanity check, to not `panic` in state_transition
 	if chainConfig.IsLondon(big.NewInt(int64(prestate.Env.Number))) {
-		if prestate.Env.BaseFee == nil {
+		if prestate.Env.ParentBaseFee != nil {
```
Comment (holiman) [inline cmd/evm/internal/t8ntool/transition.go:251]: Nitpick: `basefee > parentBasefee`, precedence-wise IMO. 
So: only if prestate.Env.BaseFee is nill should we check prestate.Env.ParentBaseFee

**Review by holiman — APPROVED** [review]
One nitpick, otherwise LGTM

File: cmd/evm/internal/t8ntool/transition.go (hunk 1 above)
Comment (lightclient) [reply cmd/evm/internal/t8ntool/transition.go]: Updated.

File: cmd/evm/testdata/25/exp.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        "logs": null,
+        "transactionHash": "0x92ea4a28224d033afb20e0cc2b290d4c7c2d61f6a4800a680e4e19ac962ee941",
+        "contractAddress": "0x0000000000000000000000000000000000000000",
+        "gasUsed": "0x5208",
+        "blockHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "transactionIndex": "0x0"
+      }
+    ],
+    "currentDifficulty": null,
+    "gasUsed": "0x5208"
```
Comment (holiman) [inline cmd/evm/testdata/25/exp.json:35]: How come there's no `currentBaseFee` here?

**Comment by lightclient** [comment]
Thanks @holiman -- Mario also mentioned this to me, yesterday afternoon. Was going to do it first thing this morning but you've beaten me to it :)

File: cmd/evm/testdata/25/exp.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        "transactionHash": "0x92ea4a28224d033afb20e0cc2b290d4c7c2d61f6a4800a680e4e19ac962ee941",
+        "contractAddress": "0x0000000000000000000000000000000000000000",
+        "gasUsed": "0x5208",
+        "blockHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "transactionIndex": "0x0"
+      }
+    ],
+    "currentDifficulty": null,
+    "gasUsed": "0x5208",
+    "currentBaseFee": "0x500"
```
Comment (lightclient) [inline cmd/evm/testdata/25/exp.json:36]: I think this should be different than `0x500` though since the parent was `0x500`?

Comment (lightclient) [reply cmd/evm/testdata/25/exp.json:36]: Yeah it's returning early in `misc.CalcBaseFee`. For some reason the JSON isn't being properly parsed.

Comment (holiman) [reply cmd/evm/testdata/25/exp.json:36]: test_cmd.go:261: (stderr:1) ParentGasUsed: 0
    test_cmd.go:261: (stderr:1) ParentGasLimit: 0

Both are zero
```golang
	if parent.GasUsed == parentGasTarget {
		return new(big.Int).Set(parent.BaseFee)
	}
```

Comment (holiman) [reply cmd/evm/testdata/25/exp.json:36]: hah
```
		ParentGasUsed    *math.HexOrDecimal64                `json:"parentGasUsed,omitempty"`
		ParentGasLimit   *math.HexOrDecimal64                `json:"parentGasUsed,omitempty"`
```

Comment (holiman) [reply cmd/evm/testdata/25/exp.json:36]: Did you manually modify an auto-generated file?

Comment (holiman) [reply cmd/evm/testdata/25/exp.json:36]: I pushed the fixed autogenerated file (`go generate execution.go`) now. I'll leave it to you to fix up failing tests :)

**Pull request merged**